packages/lib/constants.ts

```diff
@@ -39,7 +39,8 @@ export const IS_SELF_HOSTED = !(
   new URL(WEBAPP_URL).hostname.endsWith(".cal.dev") || new URL(WEBAPP_URL).hostname.endsWith(".cal.com")
 );
 export const EMBED_LIB_URL = process.env.NEXT_PUBLIC_EMBED_LIB_URL || `${WEBAPP_URL}/embed/embed.js`;
-export const IS_PRODUCTION = process.env.NODE_ENV === "production";
+export const CALCOM_ENV = process.env.CALCOM_ENV || process.env.NODE_ENV;
```

Comment:
I tend to like having apps fail fast, especially when it comes to configuration. This code allows for potential configuration to get mixed up in the event we don't set the CALCOM_ENV.

Ideally we would make sure CALCOM_ENV is set, otherwise an error is thrown immediately.

Is this to preserve backwards compatibility?

Reply:
@leog Bump^^

Reply:
Yes it is, so you were implying we should make it like the following?
```js
 export const CALCOM_ENV = process.env.CALCOM_ENV || "development";
```

There are places we need to rely on `process.env.CALCOM_ENV` such as `next.config.js`, which wouldn't have a default value if we forgot to set it.

Reply:
No, just wanted to confirm it was for backwards compatibility. If we change it to this and someone upgrades without seeing they need this new env var, their system will be running in "development" mode. I think what you have originally make sense.

Reply:
This does make me wonder if we need to update all of the places where we are using process.env.CALCOM_ENV to use this constant or IS_PRODUCTION instead. Reason, someone upgrades without setting this value, all of the lines of code that reference it will not work correctly.